apps/nova/src/hooks/call.ts

```diff
@@ -357,3 +357,60 @@ export function useCallEvents(
     mateVideo,
   };
 }
+
+export function useFullScreen() {
+  const intl = useFormatMessage();
+  const ref = useRef<HTMLDivElement>(null);
+  const [isFullScreen, setIsFullScreen] = useState<boolean>(false);
+
+  const onFullScreen = useCallback(() => {
+    const fullScreen = document.fullscreenElement?.id === ref.current?.id;
+    setIsFullScreen(fullScreen);
+  }, []);
+
+  const startFullScreen = useCallback(async () => {
+    try {
+      if (!ref.current) return;
+      await ref.current.requestFullscreen();
+    } catch (error) {
+      toaster.error({
+        title: intl("error.unexpected"),
+      });
+    }
+  }, []);
+
+  const exitFullscreen = useCallback(async () => {
+    if (ref.current) {
+      await document.exitFullscreen();
+    }
+  }, []);
+
+  const toggleFullScreen = useCallback(async () => {
+    if (!document.fullscreenElement) {
+      startFullScreen();
+    } else {
+      exitFullscreen();
+    }
```

Comment:
It can be cleaner by doing this. 

```suggestion
  const toggleFullScreen = useCallback(async () => {
    if (!document.fullscreenElement) return startFullScreen();
    return exitFullscreen();
```